File: 工程/Howare/Howare/View/Dialog.m
//
//  Dialog.m
//  Howare
//
//  Created by 刘向宏 on 16/3/30.
//  Copyright © 2016年 刘向宏. All rights reserved.
//

#import "Dialog.h"

@interface Dialog()
@property (nonatomic,weak) IBOutlet UIView *contentView;
@end

@implementation Dialog{
    
}

-(instancetype)initWithCoder:(NSCoder *)aDecoder{
    self = [super initWithCoder:aDecoder];
    [[NSBundle mainBundle] loadNibNamed:@"Dialog" owner:self options:nil];
    [self addSubview:self.contentView];
    return self;
}

-(void)layoutSubviews{
    [super layoutSubviews];
    self.contentView.frame = self.bounds;
}
/*
// Only override drawRect: if you perform custom drawing.
// An empty implementation adversely affects performance during animation.
- (void)drawRect:(CGRect)rect {
    // Drawing code
}
*/

@end

File: 工程/Hero/Hero/View/Dialog.swift
//
//  Dialog.swift
//  Hero
//
//  Created by 刘向宏 on 16/4/1.
//  Copyright © 2016年 刘向宏. All rights reserved.
//

import UIKit

enum DialogType {
    case Default
    case OnlyOK
    case NoBtton
}


class Dialog: UIView {

    @IBOutlet weak var contentView : UIView!
    
    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
        self.backgroundColor = UIColor.clearColor()
        //loadViews(.Default)
    }
    
    func loadViews(type : DialogType)  {
        if contentView != nil {
            contentView.removeFromSuperview()
        }
        NSBundle.mainBundle().loadNibNamed("Dialog", owner: self, options: nil)
        self.addSubview(contentView)
        layer.cornerRadius = 4
    }
    
    override func layoutSubviews() {
        super.layoutSubviews()
        if contentView != nil {
            contentView.frame = self.bounds
        }
    }
    
    /*
    // Only override drawRect: if you perform custom drawing.
    // An empty implementation adversely affects performance during animation.
    override func drawRect(rect: CGRect) {
        // Drawing code
    }
    */

}
